tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,166 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	var runningTime time.Duration = 30 // time for which operation should be performed
+	noOfPrefixes := 36                 // total number of prefixes which will be watched upon
+	watchPerPrefix := 10               // number of watchers per prefix
+	reqRate := 20                      // put request per second
+	keyPrePrefix := 100000             // max number of keyPrePrefixs for put operation
+
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string) // map for holding prefixes and values
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	keys := generateRandomKeys(10, keyPrePrefix)
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	for i := range prefixes {
+		prefix := prefixes[i]
+		// initialize prefix key map for each prefix
+		prefixKeysMap[prefix] = make([]string, 0)
```

Comment:
> In the above log if you see the key msg9d is being put assuming prefix for msg9 and key 'md' is being put for prefix m later. So for watcher with prefix m msg9d occurs first , but in our logic in the loop we are expecting md since that was the key it is being thought to be put first.
Hence what i was doing earlier was right, to put keys matching for all prefixes before performing the put operation since that's the way the watchers will behave.

A easy solution to that is to make all prefixes the same length. Is there a reason we want to make prefix variable length?